File: python/CAPSTONE_2/pages/login_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.action_chains import ActionChains
from pages.base_page import BasePage
import time


class LoginPage(BasePage):
    def __init__(self, driver):
        super().__init__(driver)
        self.url = "https://bstackdemo.com/"
        self.sign_in_button = (By.ID, "signin")
        self.username_dropdown = (By.CSS_SELECTOR, "div#username div.css-1hwfws3")  # clickable div
        self.password_dropdown = (By.CSS_SELECTOR, "div#password div.css-1hwfws3")
        self.login_button = (By.ID, "login-btn")
        self.products_container = (By.CLASS_NAME, "shelf-item")

    def login(self, username="demouser", password="testingisfun99"):
        """Perform login on BrowserStack Demo"""
        self.open(self.url)
        self.wait.until(EC.element_to_be_clickable(self.sign_in_button)).click()

        # Wait for login modal to appear
        self.wait.until(EC.visibility_of_element_located(self.username_dropdown))
        time.sleep(1)

        # Select username
        self.driver.find_element(*self.username_dropdown).click()
        username_option = self.wait.until(
            EC.element_to_be_clickable((By.XPATH, f"//div[contains(@id,'react-select-2-option-0-0') or text()='{username}']"))
        )
        ActionChains(self.driver).move_to_element(username_option).click().perform()

        # Select password
        self.driver.find_element(*self.password_dropdown).click()
        password_option = self.wait.until(
            EC.element_to_be_clickable((By.XPATH, f"//div[contains(@id,'react-select-3-option-0-0') or text()='{password}']"))
        )
        ActionChains(self.driver).move_to_element(password_option).click().perform()

        # Click Login
        self.wait.until(EC.element_to_be_clickable(self.login_button)).click()

        # Wait for home page products
        self.wait.until(EC.presence_of_all_elements_located(self.products_container))
        print(f"✅ Logged in successfully as {username}")

File: selenium with java/testcase3/CAPSTONE_2 (JAVA)/src/main/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.*;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class LoginPage extends BasePage {

    // Locators
    private final String url = "https://bstackdemo.com/";
    private final By signInButton = By.id("signin");
    private final By usernameDropdown = By.cssSelector("div#username div.css-1hwfws3");
    private final By passwordDropdown = By.cssSelector("div#password div.css-1hwfws3");
    private final By loginButton = By.id("login-btn");
    private final By productsContainer = By.className("shelf-item");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    // Perform login on BrowserStack Demo
    public void login(String username, String password) {
        try {
            driver.get(url);
            wait.until(ExpectedConditions.elementToBeClickable(signInButton)).click();
            System.out.println("🟢 Clicked Sign In button.");

            // Wait for dropdowns to appear
            wait.until(ExpectedConditions.visibilityOfElementLocated(usernameDropdown));
            Thread.sleep(1000);

            // Select username
            WebElement usernameDrop = driver.findElement(usernameDropdown);
            usernameDrop.click();
            WebElement usernameOption = wait.until(
                ExpectedConditions.elementToBeClickable(
                    By.xpath("//div[contains(@id,'react-select-2-option-0-0') or text()='" + username + "']")
                )
            );
            new Actions(driver).moveToElement(usernameOption).click().perform();
            System.out.println("✅ Selected username: " + username);

            // Select password
            WebElement passwordDrop = driver.findElement(passwordDropdown);
            passwordDrop.click();
            WebElement passwordOption = wait.until(
                ExpectedConditions.elementToBeClickable(
                    By.xpath("//div[contains(@id,'react-select-3-option-0-0') or text()='" + password + "']")
                )
            );
            new Actions(driver).moveToElement(passwordOption).click().perform();
            System.out.println("✅ Selected password.");

            // Click login
            wait.until(ExpectedConditions.elementToBeClickable(loginButton)).click();
            System.out.println("🚀 Clicked Login button.");

            // Wait for product grid (homepage)
            wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(productsContainer));
            System.out.println("✅ Logged in successfully as " + username);

        } catch (Exception e) {
            System.err.println("❌ Login failed: " + e.getMessage());
            throw new RuntimeException(e);
        }
    }
}
